### PR TITLE
Feature/qts 51 login refactor

### PR DESCRIPTION
**Pull request by gjanakiram**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
1. Changed project to angular cli project and removed un-wanted configuration files and npm modules. We will add necessary configuration, modules etc.. based on our need. 
2. Refactor login functionality using Redux and added couple of jasmine and protractor tests. 

* **What is the current behavior?** (You can also link to an open issue here)
1. Login logout functionality not implemented. 

* **What is the new behavior (if this is a feature change)?**
1. Login functionality working
2. Well organized redux related files and moved common functionality to shared folder
3. working unit and basic functional test cases

* **Other information**:
@gopalakrishnanu  I Removed un-wanted bunch of files but I kept jenkins and sonar related files.  Please let me know if I miss any files that you checked in.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,174 +1,92 @@
...
-    "@angular/core": "~4.4.4",
-    "@angular/forms": "~4.4.4",
-    "@angular/http": "~4.4.4",
-    "@angular/platform-browser": "~4.4.4",
-    "@angular/platform-browser-dynamic": "~4.4.4",
-    "@angular/platform-server": "~4.4.4",
-    "@angular/router": "~4.4.4",
-    "@angularclass/hmr": "~2.1.3",
-    "@angularclass/hmr-loader": "^3.0.4",
+    "@angular/animations": "^5.0.0",
```
Comment (gopalakrishnanu) [inline package.json]: Looks like we are using angular 5 now!

File: src/app/shared/auth/auth.service.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                })
+
+  }
+
+  saveJwt(jwt: string) {
+      if(jwt) localStorage.setItem('id_token', jwt)
+  }
+
+  logout() {
+    return of(true);
```
Comment (gopalakrishnanu) [inline src/app/shared/auth/auth.service.ts]: Logout should remove the jwt token from the store.

File: src/app/shared/auth/login-form.component.html [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            Username is required
+          </div>
+        </div>
+      </div>
+    </div>
+
+    <div class="form-group">
+      <div class="input-div">
+        <i class="ml-2 mdi mdi-lock-outline mdi-24px"></i>
+        <input class="form-control form-control-login" id="inputPassword3"
```
Comment (gopalakrishnanu) [inline src/app/shared/auth/login-form.component.html]: why id=inputPassword3?

File: src/app/shared/auth/login-form.component.html [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    </div>
+    <div class="mt-2">
+      <div *ngIf="errorMessage" class="warning-2">
+        {{errorMessage}}
+      </div>
+    </div>
+    <div class="form-group">
+      <div class="input-div">
+        <i class="ml-2 mdi mdi-email-outline mdi-24px"></i>
+        <input class="form-control form-control-login" id="inputEmail3"
```
Comment (gopalakrishnanu) [inline src/app/shared/auth/login-form.component.html]: why iputEmail3? Change to username.

File: src/app/shared/auth/login-page.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
+import { TestBed, ComponentFixture } from '@angular/core/testing';
+import { MatInputModule, MatCardModule } from '@angular/material';
```
Comment (gopalakrishnanu) [inline src/app/shared/auth/login-page.component.spec.ts]: How is this test different from login-form component spec?

File: package.json (hunk 1 above)
Comment (gjanakiram) [reply package.json]: Yes Gopal. I generated this project using angular cli so added latest packages. I think we should go with latest.

File: src/app/shared/auth/auth.service.ts (hunk 2 above)
Comment (gjanakiram) [reply src/app/shared/auth/auth.service.ts]: Done. Thanks Gopal

File: src/app/shared/auth/login-form.component.html (hunk 4 above)
Comment (gjanakiram) [reply src/app/shared/auth/login-form.component.html]: This entire html file written by John & Ben. Then are still working on applying css.  They has trouble to apply this html so I did in my local then hand it over to John. Once they are ready with CSS I will give proper names.

File: package.json (hunk 1 above)
Comment (gopalakrishnanu) [reply package.json]: I am fine that. Is there any major changes between angular 4 and 5?

File: src/app/shared/auth/login-page.component.spec.ts (hunk 5 above)
Comment (gjanakiram) [reply src/app/shared/auth/login-page.component.spec.ts]: in login page spec testing if store dispatches or not. In login-form we are testing submit method called with given credentials are not.

File: package.json (hunk 1 above)
Comment (gjanakiram) [reply package.json]: Major one is performance improvement, bug fixes and more.

**Comment by gjanakiram** [comment]
@gopalakrishnanu  @shamoons Can I merge these changes into develop ?

**Comment by shamoons** [comment]
@gjanakiram looks good to me

**Pull request merged**